src/jit/gentree.h

```diff
@@ -4097,6 +4101,30 @@ inline bool GenTree::IsZero()
     return false;
 }
 
+// Note that, unlike IsZero(), this only returns true only for integral types.
+inline bool GenTree::IsIOne()
+{
+    if ((gtOper == GT_CNS_INT) && (gtIntConCommon.IconValue() == 1))
+        return true;
+
+    if ((gtOper == GT_CNS_LNG) && (gtIntConCommon.LngValue() == 1))
+        return true;
+
```

Comment:
That's why I called it "IsIOne" instead of IsOne. I can go for IsIntegralOne instead.